### ISSUE TITLE
[BUG] Can't build in openjdk 21

### ISSUE THREAD

**Issue by Fmstrat** [iBotPeaches/Apktool#3362]
### Information
1. **Apktool Version (`apktool -version`)** - `master`
2. **Operating System (Mac, Linux, Windows)** - Linux (docker, see below)
3. **APK From? (Playstore, ROM, Other)** - n/a
4. **Java Version (`java --version`)** - `sapmachine:21-jdk-headless-ubuntu-jammy`

### Steps to Reproduce
This is in relation to the PR for allowing docker builds for beta (`master` branch). 

Using: `openjdk:21-slim` (Deprecated, will be replaced in PR, just here for reference)
```
> Task :brut.apktool:apktool-cli:proguard FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':brut.apktool:apktool-cli:proguard'.
> java.io.IOException: Can't read [/usr/local/openjdk-21/jmods(;;;;;;;!**.jar;!module-info.class)] (Can't read [java.security.sasl.jmod] (Can't process class [com/sun/security/sasl/ClientFactoryImpl.class] (Unsupported version number [65.0] (maximum 64.65535, Java null))))
```

Using `sapmachine:21-jdk-headless-ubuntu-jammy` or v17:
```
> Task :brut.apktool:build

> Task :brut.j.common:compileJava
/build/brut.j.common/src/main/java/brut/common/TraversalUnknownFileException.java:2: error: unmappable character (0xC5) for encoding US-ASCII
 *  Copyright (C) 2010 Ryszard Wi??niewski <brut.alll@gmail.com>
                                 ^
/build/brut.j.common/src/main/java/brut/common/TraversalUnknownFileException.java:2: error: unmappable character (0x9B) for encoding US-ASCII
 *  Copyright (C) 2010 Ryszard Wi??niewski <brut.alll@gmail.com>
                                  ^
/build/brut.j.common/src/main/java/brut/common/BrutException.java:2: error: unmappable character (0xC5) for encoding US-ASCII
 *  Copyright (C) 2010 Ryszard Wi??niewski <brut.alll@gmail.com>
                                 ^
/build/brut.j.common/src/main/java/brut/common/BrutException.java:2: error: unmappable character (0x9B) for encoding US-ASCII
 *  Copyright (C) 2010 Ryszard Wi??niewski <brut.alll@gmail.com>
                                  ^
/build/brut.j.common/src/main/java/brut/common/RootUnknownFileException.java:2: error: unmappable character (0xC5) for encoding US-ASCII
 *  Copyright (C) 2010 Ryszard Wi??niewski <brut.alll@gmail.com>
                                 ^
/build/brut.j.common/src/main/java/brut/common/RootUnknownFileException.java:2: error: unmappable character (0x9B) for encoding US-ASCII
 *  Copyright (C) 2010 Ryszard Wi??niewski <brut.alll@gmail.com>
                                  ^
/build/brut.j.common/src/main/java/brut/common/InvalidUnknownFileException.java:2: error: unmappable character (0xC5) for encoding US-ASCII
 *  Copyright (C) 2010 Ryszard Wi??niewski <brut.alll@gmail.com>
                                 ^
/build/brut.j.common/src/main/java/brut/common/InvalidUnknownFileException.java:2: error: unmappable character (0x9B) for encoding US-ASCII
 *  Copyright (C) 2010 Ryszard Wi??niewski <brut.alll@gmail.com>
                                  ^

> Task :brut.j.common:processResources NO-SOURCE
> Task :brut.j.common:classes
> Task :brut.j.common:jar

> Task :brut.j.common:javadoc FAILED
/build/brut.j.common/src/main/java/brut/common/TraversalUnknownFileException.java:2: error: unmappable character (0xC5) for encoding US-ASCII
 *  Copyright (C) 2010 Ryszard Wi??niewski <brut.alll@gmail.com>
                                 ^
/build/brut.j.common/src/main/java/brut/common/TraversalUnknownFileException.java:2: error: unmappable character (0x9B) for encoding US-ASCII
 *  Copyright (C) 2010 Ryszard Wi??niewski <brut.alll@gmail.com>
                                  ^
/build/brut.j.common/src/main/java/brut/common/BrutException.java:2: error: unmappable character (0xC5) for encoding US-ASCII
 *  Copyright (C) 2010 Ryszard Wi??niewski <brut.alll@gmail.com>
                                 ^
/build/brut.j.common/src/main/java/brut/common/BrutException.java:2: error: unmappable character (0x9B) for encoding US-ASCII
 *  Copyright (C) 2010 Ryszard Wi??niewski <brut.alll@gmail.com>
                                  ^
/build/brut.j.common/src/main/java/brut/common/RootUnknownFileException.java:2: error: unmappable character (0xC5) for encoding US-ASCII
 *  Copyright (C) 2010 Ryszard Wi??niewski <brut.alll@gmail.com>
                                 ^
/build/brut.j.common/src/main/java/brut/common/RootUnknownFileException.java:2: error: unmappable character (0x9B) for encoding US-ASCII
 *  Copyright (C) 2010 Ryszard Wi??niewski <brut.alll@gmail.com>
                                  ^
/build/brut.j.common/src/main/java/brut/common/InvalidUnknownFileException.java:2: error: unmappable character (0xC5) for encoding US-ASCII
 *  Copyright (C) 2010 Ryszard Wi??niewski <brut.alll@gmail.com>
                                 ^
/build/brut.j.common/src/main/java/brut/common/InvalidUnknownFileException.java:2: error: unmappable character (0x9B) for encoding US-ASCII
 *  Copyright (C) 2010 Ryszard Wi??niewski <brut.alll@gmail.com>
                                  ^
8 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':brut.j.common:javadoc'.
> Javadoc generation failed. Generated Javadoc options file (useful for troubleshooting): '/build/brut.j.common/build/tmp/javadoc/javadoc.options'
```

Looks like `javac` needs an `--encoding` flag, or maybe a required package for locale? I'm not super experienced with this.

**Comment by iBotPeaches** [comment]
Thanks.

21 is not yet supported in terms of CI (https://github.com/iBotPeaches/Apktool/blob/master/.github/workflows/build.yml#L72). So I haven't had a chance to peek what might be needed. Normally JDK upgrades require Gradle, plugins and more.

**Comment by iBotPeaches** [comment]
Blocking till Gradle supports 21 - https://docs.gradle.org/current/userguide/compatibility.html

**Comment by Fmstrat** [comment]
Hmm. It's merged for 8.4 https://github.com/gradle/gradle/issues/25573

Perhaps I should drop the container to J8 for now.

**Comment by iBotPeaches** [comment]
We upgrade Gradle versions quite quickly, so once its gold we can move on it. Saves me some extra troubles when I only focus versions I know officially are supported with the tool chain.

The container could probably be one of the LTS versions - 8, 11 or 17.
